### PR TITLE
FSA-547 add limited hooks

### PR DESCRIPTION
**Pull request by webecks**
[FSA-547](https://humanmade.atlassian.net/browse/FSA-547) - Separate videos from current sitemap into its own sitemap

This is related to https://github.com/humanmade/fox-sports-asia/pull/995

I only need to add a parameter to the original code to support multi post type.
**For example in msm-sitemap.php:**
- `Metro_Sitemap::date_range_has_posts( $date_stamp, $date_stamp, $post_type )` - (line 154) to be called in wp-cli.php
- `Metro_Sitemap::generate_sitemap_for_date( $date_stamp, $post_type )` - (line 156) to be called in wp-cli.php
- changed from `date_range_has_posts( $start_date, $end_date )` (line 384) -> `date_range_has_posts( $start_date, $end_date, $post_type = '' )`.
- changed from `$post_types_in = self::get_supported_post_types_in()` (line 390) -> `$post_types_in = empty( $post_type ) ? self::get_supported_post_types_in() : $wpdb->prepare( '%s', $post_type )`
- and more like this, by adding a parameter $post_type

I could not find a way to add hooks and filters around this.
If you have other ideas, please let me know. Thanks.

**There are 2 sitemaps generated from this code adjustment:**
- sitemap.xml
- sitemap-videos.xml

*I have:*
- [x] Written doc blocks and inline documentation on all new changes
- [x] Added the Jira reference as a prefix to the title of the PR
- [ ] Notes on how-to-test supplied in issue description on Jira
- [ ] WP CLI commands supplied in issue description on Jira
- [ ] As a developer, I've tested this on www-development.foxsportsasia.com

### CODE REVIEW

**Comment by tareiking** [comment]
@webecks - was this meant to be opened against the `fs-master` branch? If so, can you edit the PR and change the base?

**Comment by webecks** [comment]
@tareiking I changed the base.
There were 2 conflicts between fs-master and branch FSA-547-split-sitemap-by-posttype.
I fixed and merged it. But just realized that it might need to be merged by someone else (perhaps you) after the code changes are approved.
Could you please check? Thanks.

File: msm-sitemap.php [hunk 1]
```diff
@@ -89,6 +94,8 @@ public static function sitemap_rewrite_init() {
 		} else {
 			add_rewrite_rule( '^sitemap(\-.*)?.xml$', 'index.php?sitemap=true&type=$matches[1]', 'top' );
 		}
+
+		do_action( 'msm_sitemap_rewrite_rule' );
```
Comment (tareiking) [inline msm-sitemap.php:105]: Need a docblock to explain this action here.

**Review by tareiking — CHANGES_REQUESTED** [review]
@webecks - you'll need to add the docblock. Otherwise, looks okay for usage on Fox ATM.

**Review by webecks — COMMENTED** [review]
@tareiking is the current docblock okay?

File: msm-sitemap.php [hunk 2]
```diff
@@ -699,8 +715,8 @@ public static function build_root_sitemap_xml( $year = false ) {
 	 */
 	public static function build_sitemap_url( $sitemap_date ) {
 		$sitemap_time = strtotime( $sitemap_date );
-		$post_type = get_query_var( 'type' );
-		$post_type = empty( $post_type ) ? '' : $post_type;
+		$post_type = get_query_var( 'post_type' );
```
Comment (tareiking) [inline msm-sitemap.php:716]: We should check if this post type is registered as well here using `post_type_exists`.

File: msm-sitemap.php [hunk 3]
```diff
@@ -80,15 +85,26 @@ public static function sitemap_init() {
...
+
+		/**
+		 * Setup additional rewrite rule for the sitemap.
+		 *
+		 * Do add_rewrite_rule().
+		 * Don't forget to add 'redirect_canonical' hook for new rule. {@see 'redirect_canonical'}
+		 *
+		 * @since 1.3.0
+		 *
+		 * @param type  $var Description.
```
Comment (tareiking) [inline msm-sitemap.php]: This `param` isn't actually valid here.

**Review by tareiking — CHANGES_REQUESTED** [review]
@webecks tested locally, this creates the sitemaps as expected. Added a few comments.. getting close!

**Comment by webecks** [comment]
@tareiking I have updated the file.

**Review by tareiking — APPROVED** [review]
Approving recent changes, thanks @webecks

**Pull request merged**